animation_nodes/nodes/newnodes/gpencils_stroke_line_width.py

```diff
@@ -0,0 +1,58 @@
+import bpy
+from ... data_structures import Stroke
+from ... base_types import AnimationNode, VectorizedSocket
+
+class GPencilStrokeLineWidthNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPencilStrokeLineWidthNode"
+    bl_label = "GPencil Stroke Line Width"
+    bl_width_default = 165
+
+    useStrokeList: VectorizedSocket.newProperty()
+    useFloatList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput(VectorizedSocket("Stroke", "useStrokeList",
+            ("Stroke", "stroke"), ("Strokes", "strokes")))
+        self.newInput(VectorizedSocket("Float", "useFloatList",
+            ("Line Width", "lineWidth"), ("Line Widths", "lineWidths")), value = 100)
+        self.newOutput(VectorizedSocket("Stroke", "useStrokeList",
+            ("Stroke", "outStroke"), ("Strokes", "outStrokes")))
+
+    def getExecutionFunctionName(self):
+        if self.useStrokeList and self.useFloatList:
+            return "executeListList"
+        elif self.useStrokeList:
+            return "executeList"
+        else:
+            return "executeSingle"
+
+    def executeSingle(self, stroke, lineWidth):
+        if stroke is None: return None
+        outStroke = self.copyStroke(stroke)
+        outStroke.line_width = lineWidth
+        return outStroke
+
+    def executeList(self, strokes, lineWidth):
+        if len(strokes) == 0: return strokes
+        outStrokes = []
+        for stroke in strokes:
+            if stroke is not None:
+                outStroke = self.copyStroke(stroke)
+                outStroke.line_width = lineWidth
+                outStrokes.append(outStroke)    
+        return outStrokes
+
+    def executeListList(self, strokes, lineWidths):
+        if len(strokes) == 0 or len(lineWidths) == 0 or len(strokes) != len(lineWidths): return strokes
+        outStrokes = []
+        for i, stroke in enumerate(strokes):
+            if stroke is not None:
+                outStroke = self.copyStroke(stroke)
+                outStroke.line_width = lineWidths[i]
+                outStrokes.append(outStroke)
+        return outStrokes
+
+    def copyStroke(self, stroke):
```

Comment:
Just a quick note: you probably don't have to copy the stroke yourself every time. AN has builtin functionality that makes copies when necessary. Check how it is done for e.g. meshes or lists.

You need to define a copy expression in the socket, so that AN knows how to copy your type. Then, when a socket input is modified, you can use the `dataIsModified` property to tell AN that this input will be modified. It's not a perfect solution, but it has less overhead then your current approach.

Reply:
Thanks, @JacquesLucke for the suggestion. Initially, I added the copy option to the socket but it was not working probably I was missing the "dataIsModified" option. I'll try my best to add this in gpencil node codes. Sorry for the late reply.